### ISSUE TITLE
[raet] run-time exception: 'SaltSchedule' object has no attribute 'schedule'

### ISSUE THREAD

**Issue by plastikos** [saltstack/salt#21640]
The following non-fatal back-trace is dumped when `salt-master` is started:
```python
$> /home/omniture/install/bin/salt-master -c /home/omniture/install/etc/salt
Process Process-1:
Traceback (most recent call last):
  File "/home/omniture/install/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/omniture/install/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/omniture/install/lib/python2.7/site-packages/salt/daemons/flo/maint.py", line 64, in _maint
    consolepath=consolepath,
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/app/run.py", line 131, in run
    skedder.run()
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/base/skedding.py", line 273, in run
    status = tasker.runner.send(tasker.desire)
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/base/framing.py", line 612, in makeRunner
    self.recur() #.desire may change here
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/base/framing.py", line 425, in recur
    frame.recur()
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/base/framing.py", line 1233, in recur
    act()
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/base/acting.py", line 98, in __call__
    return (self.actor(**self.parms))
  File "/home/omniture/install/lib/python2.7/site-packages/ioflo/base/acting.py", line 460, in __call__
    return self.action(**kwa)
  File "/home/omniture/install/lib/python2.7/site-packages/salt/daemons/flo/core.py", line 633, in action
    self.schedule.eval()
AttributeError: 'SaltSchedule' object has no attribute 'schedule'
```
Installed software versions:
```
libsodium-1.0.2-0.omtr.1.i386
python2.7-enum34-1.0.4-1.noarch
python2.7-ioflo-1.2.1-1.noarch
python2.7-libnacl-1.4.1-1.noarch
python2.7-raet-0.6.3-1.noarch
salt-2015.2.0rc2015.03.13-0.omtr.1.noarch
salt-master-2015.2.0rc2015.03.13-0.omtr.1.noarch
salt-minion-2015.2.0rc2015.03.13-0.omtr.1.noarch
```
This has `RAET` enabled:
`transport: raet`

**Comment by plastikos** [comment]
A potentially similar failure happens in `salt-minion`:
```
$> /home/omniture/install/bin/salt-minion -c /home/omniture/install/etc/salt
[WARNING ] Although 'dmidecode' was found in path, the current user cannot execute it. Grains output might not be accurate.
[ERROR   ] 'SaltRaetRoadStackJoiner' object has no attribute 'masters'
[WARNING ] ** Restarting minion **
```

It appears that both of these problems occur because the initialization of
a class attribute in the `postinitio()` method does not take place prior to
the `action()` method being invoked.  Unfortunately I do not have
background in `ioflo` so I'm unable to debug this further.

Is the `python-ioflo` version `1.2.1` too new?

**Comment by jfindlay** [comment]
@plastikos, thanks for the report.  I have a limited knowledge of raet and ioflo currently, but it seems you've found a legitimate issue.

**Comment by plastikos** [comment]
Here's the `test.versions_report`:
```
httpd@qe2.qc ~][QC]$ salt-call --local test.versions_report
[WARNING ] Although 'dmidecode' was found in path, the current user cannot execute it. Grains output might not be accurate.
local:
               Salt: 2015.2.0
             Python: 2.7.8 (default, Aug  7 2014, 06:25:46)
             Jinja2: 2.7.1
           M2Crypto: 0.21.1
     msgpack-python: 0.4.0
       msgpack-pure: Not Installed
           pycrypto: 2.6
            libnacl: 1.4.1
             PyYAML: 3.11
              ioflo: 1.2.1
              PyZMQ: 14.0.1
               RAET: 0.6.3
                ZMQ: 3.2.5
               Mako: 0.9.1
```

**Comment by double-yaya** [comment]
I have the same question,when I restart salt-minion ,happend like this

salt --versions-report
           Salt: 2014.7.2
         Python: 2.6.6 (r266:84292, May  1 2012, 13:52:17)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.1.13
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: 1.4.2
         PyYAML: 3.10
          ioflo: 1.2.1
          PyZMQ: 2.2.0.1
           RAET: 0.4.2
            ZMQ: 3.2.4
           Mako: Not Installed

**Comment by Snergster** [comment]
seeing the same:  call it a p3 or medium if you like but from my pov raet is completely busted right now
[INFO    ] The salt minion is shut down
[ERROR   ] 'SaltRaetRoadStackJoiner' object has no attribute 'masters'
[WARNING ] ** Restarting minion **


salt-minion --versions-report
           Salt: 2015.5.0
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.22
 msgpack-python: 0.4.5
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: 1.4.2
         PyYAML: 3.10
          ioflo: 1.2.1
          PyZMQ: 14.0.1
           RAET: 0.6.3
            ZMQ: 4.0.4
           Mako: 0.9.1

**Comment by karlsoon** [comment]
Same problem.
In logs:
2015-05-12 21:54:02,287 [salt.scripts     ][ERROR   ][24015] 'SaltRaetRoadStackJoiner' object has no attribute 'masters'
2015-05-12 21:54:02,287 [salt.scripts     ][WARNING ][24015] ** Restarting minion **

master:/etc/salt/minion.d # salt --versions-report  
           Salt: 2015.5.0
         Python: 2.7.9 (default, Dec 13 2014, 18:02:08) [GCC]
         Jinja2: 2.7.3
       M2Crypto: 0.22
 msgpack-python: 0.4.6
   msgpack-pure: 0.1.3
       pycrypto: 2.6.1
        libnacl: 1.4.2
         PyYAML: 3.11
          ioflo: 1.2.1
          PyZMQ: 14.6.0
           RAET: 0.6.3
            ZMQ: 4.0.5
           Mako: 1.0.1

Process Process-1:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/salt/cli/caller.py", line 298, in minion_run
    minion = salt.Minion()  # daemonizes here
AttributeError: 'module' object has no attribute 'Minion

**Comment by DmitryKuzmenko** [comment]
The first bunch of issues are related to the ioflo >= 1.2.0 release which dropped deprecated postinitio support. This method was renamed to _prepare. This issue is fixed in develop branch with 1a9ff9928a7393a6312354e7222d456b35a1b605. Also found one similar, sent PR.

This last one is related to transport layer redesign so salt module doesn't contain Minion class anymore. This issue is also already fixed in develop with 6b81e2a38f5427e2cb6fcb354b5dae7f3ec0b3b4.
